tests/plugins/test_add_help.py

```diff
@@ -240,3 +243,55 @@ def check_popen_fail(*args, **kwargs):
         assert list(result.keys()) == ['add_help']
         assert isinstance(result['add_help'], RuntimeError)
         assert 'Error running' in str(result['add_help'])
+
+
+@pytest.mark.parametrize('filename', ['help.md', 'other_file.md'])  # noqa
+def test_add_help_generate_metadata(tmpdir, docker_tasker, filename):
+    df_content = """FROM fedora
+LABEL name='test' \\
+      maintainer='me'
+"""
+
+    df = df_parser(str(tmpdir))
+    df.content = df_content
+
+    workflow = DockerBuildWorkflow(MOCK_SOURCE, 'test-image')
+    workflow.builder = X
+    workflow.builder.df_path = df.dockerfile_path
+    workflow.builder.df_dir = str(tmpdir)
+
+    help_markdown_path = os.path.join(workflow.builder.df_dir, filename)
+    generate_a_file(help_markdown_path, "foo")
+    help_man_path = os.path.join(workflow.builder.df_dir, AddHelpPlugin.man_filename)
+    generate_a_file(help_man_path, "bar")
+
+    cmd = ['go-md2man', '-in={}'.format(help_markdown_path), '-out={}'.format(help_man_path)]
+
+    def check_popen(*args, **kwargs):
+        assert args[0] == cmd
+        return MockedPopen()
+
+    (flexmock(subprocess)
+     .should_receive("Popen")
+     .once()
+     .replace_with(check_popen))
+
+    runner = PreBuildPluginsRunner(
+        docker_tasker,
+        workflow,
+        [{
+            'name': AddHelpPlugin.key,
+            'args': {'help_file': filename}
+        }]
+    )
+    runner.run()
+    lines = ""
+    with open(help_markdown_path) as f:
+        lines = "".join(f.readlines())
+
+    example = """%% test (1) Container Image Pages
```

Comment:
nitpick: wrap the text in `textwrap.dedent` to preserve identation (example: https://github.com/projectatomic/atomic-reactor/blob/master/tests/plugins/test_add_filesystem.py#L144)

Reply:
I did this intentionally because I wanted to workarout this situation
```
dedent(
        """        %% test (1) Container Image Pages       <--  this line as it should contain exact amount of spaces
        %% me
        %% %s
        foo""")
```
but if you want this solution there is no problem to fix that :)

Reply:
>this line as it should contain the exact amount of spaces prefixing the real content

Not sure if I understood this correctly - the spaces must be there? If yes the text is not explicit about it.

>but I don't want empty line at start of the string
Use `"""\` to avoid adding a new line

Its a minor thing anyway, I'm fine with current implementation, unless the first line has to contain spaces - this might be important